ui/shared/tabs/failureSummary/FailureSummaryTab.jsx

```diff
@@ -67,6 +89,30 @@ class FailureSummaryTab extends React.Component {
     window.open(data.url);
   };
 
+  internalIssueFilerCallback = async (data) => {
+    const { addBug } = this.props;
+    const { suggestion, suggestions } = this.state;
+
+    await addBug({ ...data, newBug: `i${data.internal_id}` });
+    window.dispatchEvent(new CustomEvent(thEvents.saveClassification));
+
+    // Try matching an internal bug already fetched with enough occurences
+    const internalBugs = suggestions
+      .map((s) => s.bugs.open_recent)
+      .flat()
+      .filter((bug) => bug.id === null);
+    const existingBug = internalBugs.filter(
+      (bug) => bug.internal_id === data.internal_id,
+    )[0];
+    // Check if we reached the required number of occurrence to open a bug in Bugzilla
```

Comment:
Can this be used when a new classification with an internal issue gets created with pinning (instead of 'creating' an internal issue again like here)?

Reply:
It is much harder to implement because we are not sure the bug suggestions are available (we are out of the scope of the `FailureSummaryTab`). We could however support the case where the bug is part of the actual suggestions (thus reusing the above code). This is probably doable via a custom event.